File: lib/tmuxinator/doctor.rb
module Tmuxinator
  class Doctor
    class << self
      def editor?
        !ENV["EDITOR"].nil? && !ENV["EDITOR"].empty?
      end

      def installed?
        Kernel.system("type tmux > /dev/null")
      end

      def shell?
        !ENV["SHELL"].nil? && !ENV["SHELL"].empty?
      end
    end
  end
end

File: tmuxinator-go/internal/doctor/doctor.go
// Package doctor provides environment validation for tmuxinator.
package doctor

import (
	"os"
	"os/exec"
)

// Installed returns true if tmux is available in PATH.
func Installed() bool {
	_, err := exec.LookPath("tmux")
	return err == nil
}

// EditorSet returns true if $EDITOR is set and non-empty.
func EditorSet() bool {
	e := os.Getenv("EDITOR")
	return e != ""
}

// ShellSet returns true if $SHELL is set and non-empty.
func ShellSet() bool {
	s := os.Getenv("SHELL")
	return s != ""
}

